src/main/java/org/reaktivity/nukleus/http2/internal/Http2Stream.java

```diff
@@ -115,9 +116,15 @@ void onHttpReset()
             factory.doReset(applicationReplyThrottle, applicationReplyId, 0);
         }
 
-        connection.writeScheduler.rst(http2StreamId, Http2ErrorCode.CONNECT_ERROR);
-
-        factory.counters.resetStreamFramesWritten.getAsLong();
+        if (factory.correlations.containsKey(correlationId))
+        {
```

Comment:
We need to proactively remove the correlation from the map and detect when it is non-null to avoid a potential memory leak in the correlations map.